common/web/gesture-recognizer/src/engine/headless/inputEngineBase.ts

```diff
@@ -20,17 +23,74 @@ interface EventMap<HoveredItemType, StateToken> {
 export abstract class InputEngineBase<HoveredItemType, StateToken = any> extends EventEmitter<EventMap<HoveredItemType, StateToken>> {
   private _activeTouchpoints: GestureSource<HoveredItemType>[] = [];
 
+  // Touch interactions in the browser actually _re-use_ touch IDs once they lapse; the IDs are not lifetime-unique.
+  // This gesture-engine desires lifetime-unique IDs, though, so we map them within this engine to remedy that problem.
+  private readonly identifierMap: Record<number, number> = {};
```

Comment:
Why not use a [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) here?

Reply:
If memory serves, Android 5.0 uses Chrome 37.  Note the MDN documentation - Chrome 38 required.  We're off by a single version, but that does matter for the oldest devices we aim to support.

Granted, we _do_ polyfill for main Web, which is where this code would land... so I guess I can change it over if needed?

That said, `Record` is a special TS type... that basically allows using raw objects as if they were a `Map`.  Refer to https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type.  I've gotten quite accustomed to using that at this point; this is not the only use of it in the KMW codebase.  From my perspective, `Map` is effectively "syntactic sugar" for the same pattern.

Reply:
I saw you mentioned in one of the other comments that `Map` can be faster than use of a raw object?  I wasn't aware of that, so I went looking at references...

https://www.zhenghao.io/posts/object-vs-map

> According to [my benchmarks](https://www.zhenghao.io/posts/object-vs-map#performance-extravaganza), unless the keys are strings of small integers, Map is indeed more performant than Object on insertion, deletion and iteration speed, and it consumes less memory than an object of the same size.

I should probably point out that due to how browsers manage `Touch` identifiers... our keys here, during actual use, will consistently be "small integers". That said...

> Use Map for dictionaries or hash maps where you have a variable number of entries, with frequent updates, whose keys might not be known at author time, such as an [event emitter](https://github.com/developit/mitt/blob/main/src/index.ts#L45).

Yeah, the object's fields will vary as entries come in and out.

I'm not terribly concerned about a major performance impact here, though - the object should only ever have as many entries as there are active touchpoints - so generally 2 or 3 at the most.  It's small, and we're only hitting the map-like uses noted in these comments once per touch event each.

I suppose we actually can start using Map, so I'll try to remember that moving forward.  At the moment, though... I'd prefer to triage changing it until later as it's pretty low-impact.